tests/unit/test_dynamodb.py

```diff
@@ -549,3 +549,154 @@ def test_read_items_schema(params, dynamodb_table: str, chunked: bool):
     }
```

Comment:
I think not a single one of our tests checks that the actual data we get back is the same as the original data frame. Should we check that in existing tests perhaps to avoid similar situations in the future?